packages/qwik-city/src/buildtime/markdown/mdx.ts

```diff
@@ -79,10 +79,14 @@ export async function createMdxTransformer(ctx: BuildContext): Promise<MdxTransf
         .replace('/', '_');
       const addImport = `import { _jsxC, RenderOnce } from '@builder.io/qwik';\n`;
       const newDefault = `
-const WrappedMdxContent = () => {
-  return _jsxC(RenderOnce, {children: _jsxC(_createMdxContent, {}, 3, null)}, 3, ${JSON.stringify(key)});
+export default function MDXContent() {
+  const key = ${JSON.stringify(key)};
+  const content = _jsxC(_createMdxContent, {}, 3, null);
+
+  return typeof MDXLayout === "undefined" ?
+    _jsxC(RenderOnce, { children: content }, 3, key) :
+    _jsxC(RenderOnce, { children: _jsxC(MDXLayout, { children: content }, 3, key)});
```

Comment:
```js
  const children = typeof MDXLayout === "undefined" ? content : _jsxC(MDXLayout, { children: content }, 3)
  return _jsxC(RenderOnce, children, 3, key);
```

Reply:
I'm not 100% sure tbh. I do know the previous revision worked and so does the one with your suggestion, so I'll take the latter due to my lack of understanding of the rendering nuances of `_jsxC`. Thanks!